lib/iris/fileformats/pp.py

```diff
@@ -1297,6 +1301,16 @@ def core_data(self):
         return self._data
 
     @property
+    def data_dtype(self):
+        return self._data.dtype \
+            if self._data_dtype is None \
+            else self._data_dtype
```

Comment:
Look closer. The first line is the `dtype` attribute of `self._data`. Maybe it's a bit confusing...